### ISSUE TITLE
The alignment looks bad while loss decreases

### ISSUE THREAD

**Issue by liyz15** [keithito/tacotron#75]
Thanks for your nice code.
I'm working with mandarin. The input text have been converted like 'jin1 tian1 tian1 qi4 zhen1 bu2 cuo4'.
I've added '1234' to the _characters string and set cleaners=basic_cleaners.
Also I've checked that outputting spectrograms are correct.

However after 41k step the alignment doesn't seem to converge, it looks like this:
![step-41000-align](https://user-images.githubusercontent.com/32741741/32815893-2a5f8218-c9f1-11e7-8d35-58f6bc92435a.png)

Could it be not enough data? The dataset I use is only 2 hours approximately.

Any help will be appreciated.

**Comment by liyz15** [comment]
Here is my log file:
[train.log](https://github.com/keithito/tacotron/files/1473252/train.log)

**Comment by keithito** [comment]
Thanks for the detailed report! Unfortunately, I don't have a good answer for you. As far as I can tell, you seem to be doing everything right.
- It's possible that it's just hard to learn an alignment on your training data, but if you run for more iterations it will work.
- It's also possible that 2 hours isn't enough training data.
- If you haven't done so already, try trimming leading and trailing silence on your audio samples.
- Also, make sure there's not anything that could be interfering with learning the alignment in the training data -- background noise, large variation in volume between examples, mislabeled examples, etc.
- It might be interesting to try running some examples through eval.py or demo_server.py. The model is apparently learning _something_ because the loss is going down. Listening to the output might give you an idea of what that something is.

**Comment by liyz15** [comment]
I trained again on the same dataset for 32k steps, the alignment still looks meaningless.
However I ran eval.py and found that:
- The first word of output audio sounds like human voice, even though it may be incorrect, and after that it begins to repeat one meaningless syllable.
- No matter how long the input text is, the output audio duration is always around 19s.

Is there any possible reason for this?
Thank you very much.

**Comment by keithito** [comment]
The first issue sounds like it's due to the failure of the model to learn to attend to the input sequence when generating the output (i.e. the poor alignment). This would lead the model to generate output that sounds like speech but is not conditioned on the input.

The second issue is a result of the first one. When everything is working correctly, the model will generate silence when it is done generating output, and a post-processing step trims that silence off. If it never makes it to the end, then it will not get trimmed, and the output will always be the maximum length.

**Comment by liyz15** [comment]
I changed the dataset to a another with ~28hours length, although it's multi-speaker, I chose those speakers who have similar voice.
Now the alignment looks so much better, and the output audios also sound intelligble.
![step-44000-align](https://user-images.githubusercontent.com/32741741/33892345-866f0d82-df93-11e7-91e1-86b472c3b1fb.png)
Thanks again for your excellent work and patience!

**Comment by begeekmyfriend** [comment]
Hey, what is the data set did you use? I have tried `thchs30` with small set and the alignment seems bad too.
